src/algs/ags/ags.cc

```diff
@@ -0,0 +1,108 @@
+// A C-callable front-end to the AGS global-optimization library.
+//  -- Vladislav Sovrasov
+
+#include "ags.h"
+#include "solver.hpp"
+#include <iostream>
+#include <cstring>
+#include <exception>
+
+double ags_eps = 0;
+double ags_r = 3;
+double eps_res = 0.001;
+unsigned evolvent_density = 12;
+int ags_refine_loc = 0;
+int ags_verbose = 0;
+
+int ags_minimize(unsigned n, nlopt_func func, void *data, unsigned m, nlopt_constraint *fc,
+                 double *x, double *minf, const double *l, const double *u, nlopt_stopping *stop)
+{
+  int ret_code = NLOPT_SUCCESS;
+
+  if (n > ags::solverMaxDim)
+    return NLOPT_INVALID_ARGS;
+  if(m != nlopt_count_constraints(m, fc) || m > ags::solverMaxConstraints)
+    return NLOPT_INVALID_ARGS;
+
+  if (ags_verbose && n > 5)
+    std::cout << "Warning: AGS is unstable when dimension > 5" << std::endl;
+
+  std::vector<double> lb(l, l + n);
+  std::vector<double> ub(u, u + n);
+  std::vector<ags::NLPSolver::FuncPtr> functions;
+  for (unsigned i = 0; i < m; i++)
+  {
+    if (fc[i].m != 1)
+      return NLOPT_INVALID_ARGS;
+    functions.push_back([fc, data, n, i](const double* x) {
+      double val = 0;
+      nlopt_eval_constraint(&val, NULL, &fc[i], n, x);
+      return val;
+    });
+  }
+  functions.push_back([func, data, n, stop](const double* x) {
+    ++ *(stop->nevals_p);
+    return func(n, x, NULL, data);});
+
+  ags::SolverParameters params;
+  params.r = ags_r;
+  params.itersLimit = stop->maxeval != 0 ? stop->maxeval : 5000;
```

Comment:
This is wrong — `maxeval == 0` denotes no maximum, not 5000 iterations.  If you must set it to a specific value, use `INT_MAX`.